File: simpArrSum.cpp
#include <iostream>
#include <string>
#include <vector>
#include <random>

std::mt19937 RNG;
std::uniform_int_distribution<u_int8_t> singDigitDist(0,9);

int simpleArraySum(std::vector<int> arr) {
    int sum = 0;
    for (int i = 0; i < arr.size(); i++)
    {
        /* code */
        sum+=arr[i];
    }
    return sum;
    
}

std::vector<int> exArr = {1,2,3};

void printIntVect(std::vector<int> arr) {
    std::cout << "{ ";
    for (int i = 0; i < arr.size(); i++)
    {
        /* code */
        std::cout << arr[i];
        if (i+1 < arr.size()) {std::cout << ", ";}
    }
    std::cout << " }";
    
}

void printVectAndSum(std::vector<int> arr) {
    printIntVect(arr);
    std::cout << " -> " << simpleArraySum(arr) << "\n\n";
}

std::vector<int> createRandSingDigitVect(){
    std::vector<int> intVect = {};
    int num = singDigitDist(RNG);
    while (num == 0)
    {
        /* code */
        num = singDigitDist(RNG);
    }
    while (num != 0)
    {
        /* code */
        intVect.push_back(num);
        num = singDigitDist(RNG);
    }
    return intVect;
    
}

int main() {
    // printIntVect(exArr);
    // std::cout << " -> " << simpleArraySum(exArr) << "\n\n";
    printVectAndSum(exArr);
    RNG.seed(time(0));
    printVectAndSum(createRandSingDigitVect());
    printVectAndSum(createRandSingDigitVect());
    printVectAndSum(createRandSingDigitVect());
}

File: simpArrSum.js
function simpleArraySum(ar) {
    let sum = 0;
    for (let i = 0; i < ar.length; i++) {
        const num = ar[i];
        sum+=num;
    }
    return sum;
}

const arr = [1,2,3];
console.log(arr);
console.log(simpleArraySum(arr));

const rngSingDigi = () => {return Math.floor(Math.random()*10)};

function rngArr() {
    let newNum = rngSingDigi();
    while (newNum === 0) {
        newNum = rngSingDigi();
    }
    let numArr = [];
    while (newNum!=0) {
        numArr.push(newNum);
        newNum = rngSingDigi();
    }
    return numArr;
}

// console.log(rngArr());
// console.log(rngArr());
// console.log(rngArr());

const randArr1 = rngArr();
console.log(randArr1);
console.log(simpleArraySum(randArr1));
const randArr2 = rngArr();
console.log(randArr2);
console.log(simpleArraySum(randArr2));
const randArr3 = rngArr();
console.log(randArr3);
console.log(simpleArraySum(randArr3));
// const randArr2 = rngArr();
// console.log(randArr2);
// const randArr3 = rngArr();
// console.log(randArr3);